packages/graphiql/src/components/QueryEditor.tsx

```diff
@@ -83,210 +108,161 @@ export class QueryEditor extends React.Component<QueryEditorProps, {}>
     require('codemirror-graphql/jump');
     require('codemirror-graphql/mode');
 
-    const editor: CM.Editor = (this.editor = CodeMirror(this._node, {
-      value: this.props.value || '',
+    const selectElement = nodeRef.current;
+
+    editorRef.current = new CodeMirror(selectElement, {
+      value: session?.operation?.text || '',
       lineNumbers: true,
       tabSize: 2,
       mode: 'graphql',
-      theme: this.props.editorTheme || 'graphiql',
+      theme: props.editorTheme || 'graphiql',
       keyMap: 'sublime',
       autoCloseBrackets: true,
       matchBrackets: true,
       showCursorWhenSelecting: true,
-      readOnly: this.props.readOnly ? 'nocursor' : false,
+      readOnly: props.readOnly ? 'nocursor' : false,
       foldGutter: {
         minFoldSize: 4,
       },
       lint: {
-        schema: this.props.schema,
+        schema,
       },
       hintOptions: {
-        schema: this.props.schema,
+        schema,
         closeOnUnfocus: false,
         completeSingle: false,
-        container: this._node,
+        container: selectElement,
       },
       info: {
-        schema: this.props.schema,
+        schema,
         renderDescription: (text: string) => md.render(text),
         onClick: (reference: GraphQLType) =>
-          this.props.onClickReference && this.props.onClickReference(reference),
+          props.onClickReference && props.onClickReference(reference),
```

Comment:
```suggestion
          props.onClickReference?.(reference),
```

This is supported by TS. If it transpiles correctly then 🙌